src/inventory-count/infra/repository/prisma.inventory-count.adapter.ts

```diff
@@ -0,0 +1,185 @@
+/* eslint-disable @typescript-eslint/require-await */
+/* eslint-disable @typescript-eslint/no-unused-vars */
+
+import { Injectable } from '@nestjs/common';
+import {
+  InventoryCountItem as InventoryCountItemDbRecord,
+  Prisma,
+} from '@prisma/client';
+import { RepositoryBase } from 'src/core/domain/repository.base';
+import {
+  PaginatedQueryParams,
+  Paginated,
+} from 'src/core/domain/repository.interface';
+import { InventoryCountTemplateNotFoundError } from 'src/inventory-count-template/errors/inventory-count-template-not-found.error';
+import { InventoryCountMapper } from 'src/inventory-count/application/mappers/inventory-count.mapper';
+import { InventoryCountItemEntity } from 'src/inventory-count/domain/inventory-count-item.entity';
+import { InventoryCountRepositoryPort } from 'src/inventory-count/domain/inventory-count-repository.port';
+import {
+  InventoryCountEntity,
+  InventoryCountItemType,
+} from 'src/inventory-count/domain/inventory-count.entity';
+import { PrismaService } from 'src/prisma/prisma.service';
+
+const InventoryCountPrismaValidator =
+  Prisma.validator<Prisma.InventoryCountDefaultArgs>()({
+    include: { inventoryCountItems: true },
+  });
+
+export type InventoryCountDbRecord = Prisma.InventoryCountGetPayload<
+  typeof InventoryCountPrismaValidator
+>;
+
+@Injectable()
+export class PrismaInventoryCountAdapter
+  extends RepositoryBase<InventoryCountEntity>
+  implements InventoryCountRepositoryPort
+{
+  constructor(
+    private readonly prisma: PrismaService,
+    private readonly mapper: InventoryCountMapper,
+  ) {
+    super(prisma);
+  }
+
+  async create(entity: InventoryCountEntity): Promise<void> {
+    await this.prisma.$transaction(async () => {
+      const inventoryCount = await this.prisma.inventoryCount.create({
+        data: this.mapToInventoryCountDbRecord(entity),
+      });
+
+      const templateStorages =
+        await this.prisma.inventoryCountTemplate.findUnique({
+          where: { id: entity.getProps().inventoryCountTemplateId },
+          include: {
+            storages: {
+              include: {
+                storage: {
+                  include: {
+                    items: true,
+                  },
+                },
+              },
+            },
+          },
+        });
+
+      if (!templateStorages) {
+        throw new InventoryCountTemplateNotFoundError(
+          `InventoryCountTemplate with ID ${entity.getProps().inventoryCountTemplateId} not found.`,
+        );
+      }
+
+      const inventoryCountItems = templateStorages.storages.flatMap(
```

Comment:
this is also part of business logic. 
Business states: when we create inventory count, it is automatically filled with inventory count ingredients with empty (null) quantity

Reply:
your repository should only handle persistence, which is taking entity and just saving it how ever it wants